Scenes/CharacterSelection/CharacterSelection.gd

```diff
@@ -21,37 +23,61 @@ onready var SaveGame = get_node("/root/Variables").SaveGame
 func _ready():
 	var saves = SaveGame.load_all()
 	for save in saves:
-		add_character(save)
-
-func add_character(save):
-	var id = save["saveFile"]
-	var name = save.name
-	var info = save.info
-	var picture = save.picture
-	var player = save.player
+			var id = save.get("file", "unknown")
+			var meta = save.get("meta", "unknown")
+			var name = "unknown"
+			var info = "unknown"
+			if typeof(meta) == TYPE_DICTIONARY:
+				name = meta.get("name", "unknown")
+				info = meta.get("info", "unknown")
+			
+			var picture = save.get("picture", CharacterDefaults.characters.alrune.picture)
+			if typeof(meta) != TYPE_DICTIONARY:
+				picture = CharacterDefaults.characters.alrune.picture
+			var player = save
+		
+			add_character(id, name, info, picture, player)
+	
+	Core.emit_signal("scene_loaded", self)
+
+func add_character(id: String, name: String, info: String, picture: Dictionary, player: Dictionary):
 	var SelectButton = CharacterSelectionButton.instance()
+	
 	SelectButton.name = str(id)
-	get_node("Scroll/HBox/VBox/ButtonsVBox").add_child(SelectButton)
-	var ButtonInstance = get_node("Scroll/HBox/VBox/ButtonsVBox/" + str(id))
+	get_node("VBox/Scroll/HBox/VBox/ButtonsVBox").add_child(SelectButton)
+	var ButtonInstance = get_node("VBox/Scroll/HBox/VBox/ButtonsVBox/" + str(id))
 	
 	# Set Picture
 	if picture:
-		ButtonInstance.get_node("VBox/Content/HBox/CenterProfile/HBox/Profile").texture = load(picture)
-		ButtonInstance.get_node("VBox/Content/HBox/CenterProfile/HBox/Profile").flip_h = player.pic[1][0]
-		ButtonInstance.get_node("VBox/Content/HBox/CenterProfile/HBox/Profile").flip_v = player.pic[1][1]
+		ButtonInstance.get_node("VBox/Content/HBox/CenterProfile/HBox/Profile").texture = load(
+			"res://Assets/Images/Profiles/" + picture.path)
+		#ButtonInstance.get_node("VBox/Content/HBox/CenterProfile/HBox/Profile").flip_h = player.pic[1][0]
+		#ButtonInstance.get_node("VBox/Content/HBox/CenterProfile/HBox/Profile").flip_v = player.pic[1][1]
```

Comment:
The image still needs to be flipped right?

Reply:
Hmmmm, I think so

Reply:
So why is it commented out?

Reply:
I don't think profile images display correctly at all in the selection menu

Reply:
Ok it works now! :D